### PR TITLE
dev: add CI on macOS

### LINKED ISSUE #39: Add CI builds on macOS
Follow-up to #2: build zkllvm in CI on macOS, at least in one environment (os version × compiler × boost).

### PR DESCRIPTION
**Pull request by NickVolynkin**
Resolves #39

### CODE REVIEW

**Comment by makxenov** [comment]
@NickVolynkin rebased and fixed, PTAL

File: .github/workflows/build_macos.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        run: >
+          /bin/bash -c "$(curl -fsSL
+            https://raw.githubusercontent.com/Homebrew/install/master/install.sh
+          )"
+
+      - name: Run brew install
+        id: brew-install
+        run: |
+          brew install \
+            boost@1.76 \
```
Comment (NickVolynkin) [inline .github/workflows/build_macos.yml:45]: @makxenov we're installing boost on this line

File: .github/workflows/build_macos.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#            # https://github.com/MarkusJx/prebuilt-boost/blob/main/versions-manifest.json
+#            boost_version: ${{ env.BOOST_VERSION }}
+#            boost_install_dir: ${{ runner.temp }}
+#            platform_version: 11
+#            toolset: clang
+#            arch: x86
+
+      - name: Configure CMake
+        env:
+          BOOST_ROOT: "/usr/local/Cellar/boost@1.76/1.76.0_3"
```
Comment (NickVolynkin) [inline .github/workflows/build_macos.yml:80]: ... and here we tell Cmake where to look for boost

File: .github/workflows/build_macos.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+#        id: install-boost
+#        with:
+#            # A list of supported versions can be found here:
+#            # https://github.com/MarkusJx/prebuilt-boost/blob/main/versions-manifest.json
+#            boost_version: ${{ env.BOOST_VERSION }}
+#            boost_install_dir: ${{ runner.temp }}
+#            platform_version: 11
+#            toolset: clang
+#            arch: x86
+
```
Comment (NickVolynkin) [inline .github/workflows/build_macos.yml:77]:
```suggestion
```

**Review by nkaskov — APPROVED** [review]

**Pull request merged**